src/jupyterlabcontroller/services/lab.py

```diff
@@ -463,9 +461,15 @@ async def create_quota(
             )
 
     def build_namespace_quota(
-        self, lab: LabSpecification
+        self, user: UserInfo
     ) -> Optional[UserResourceQuantum]:
-        return lab.namespace_quota
+        if user.quota and user.quota.notebook:
```

Comment:
Will this let us separate namespace and pod limits in its current form?

Reply:
No, Gafaelfawr doesn't support that. If that's something we'll want, we should talk about what that would look like. I'll add that to my queue for stand-up discussion.